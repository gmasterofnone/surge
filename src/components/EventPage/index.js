import React, { Component } from 'react';
import './EventPage.css'


let uuidv4 = require("uuid/v4");

class EventPage extends Component {
  constructor() {
    super()
    this.state = {
      comments: []
    }
  }

  render() {
    return(
      <div className='event-page-container'>
        <div className='event-page'>
           
        </div>
      </div>
    )
  }
}

export default EventPage;