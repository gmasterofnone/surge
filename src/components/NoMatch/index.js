import React from 'react';
import './NoMatch.css'
import { Link } from 'react-router-dom';

import whoops from '../../assets/whoops.svg'

const NoMatch = () => {
  return(
    <div className='error-page'>
      <div className='error-container'>
        <img className='whoops' src={whoops} alt='404 page'/>
        <h1 className='error-header'>Blue screen....You broke it...</h1>
        <h3 className='error-message'>ERROR 404: CLICK <Link className='error-redirect' to='/'>HERE</Link> TO UN-BREAK</h3>
      </div>
    </div>
  )
}

export default NoMatch;