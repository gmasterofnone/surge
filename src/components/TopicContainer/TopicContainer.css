.feature-event {
  width: 100%;
  height: 600px;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

.topic-container {
  margin: 20px;
  grid-column-gap: 10px;
  grid-row-gap: 15px;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr))
}
