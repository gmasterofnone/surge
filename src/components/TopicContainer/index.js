import React from 'react';
import './TopicContainer.css'
import { randomNumber } from '../../utils/Helper'

import { Event } from '../../containers/Event'
let uuidv4 = require("uuid/v4");

const TopicContainer = ( { content } ) => {
  const feature = content[0]
  const childEvents = content.filter(event => event.title !== feature.title)
  const events = childEvents.map(event => (
    <Event key={uuidv4()} event={event} /> 
  ))

  const RandomFeature = randomNumber(0, 1)
  const { topic, source, author, title, image, date, body, comments} = feature;
  const cleanedTitle = title.split('-')
  
  const avatars = comments.map(comment => (
    <img className={`avatars`} 
      src={comment.avatar} 
      alt={comment.name} 
      key={uuidv4()} 
    />
  ))
    
  return(
    <div className='topic-container'>
      <div 
        className='feature-event'
        style={{backgroundImage: `url(${image})`}}
      >
        <div className='feature-content-container'>
          <div className='feature-content'>
            <p className='feature-source'>{source.toUpperCase()} | <span>{date.toUpperCase()}</span></p>
            <h1 className='feature-title'>{cleanedTitle[0]}</h1>
            <p className='feature-body'>{body}</p>
            <div className='comment-avatars'>
              { avatars.slice(0, 3) }
              <a className='comment-count'>{`${avatars.length} comments`}</a>
            </div>
          </div>
          
        </div>
      </div>
      <div className='event-container'>
        { events }
      </div>
    </div>
  )
}



export default TopicContainer;