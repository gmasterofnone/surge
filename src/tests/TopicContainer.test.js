import React from 'react';
import { TopicContainer } from '../components/TopicContainer';
import { mount } from 'enzyme';


describe('TopicContainer', () => {
  let wrapper;
  let mockArticle;
  let styleSheet;

  beforeEach(() => {
 
    mockArticle = [{
      source: 'hey',
      comments: []
    }]
   
    wrapper = mount(
      <TopicContainer
        content={mockArticle}
      />
    )
  })

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})