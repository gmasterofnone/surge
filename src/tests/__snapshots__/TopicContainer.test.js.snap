// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicContainer should match the snapshot 1`] = `undefined`;
